### ISSUE TITLE
Throwing NotFoundException results in 500 error page

### ISSUE THREAD

**Issue by toebu** [k-next/kirby#6466]
## Description
When throwing a `Kirby\Exception\NotFoundException`, an error page is rendered as if an internal server error would have happened with an http status code of 500.

**Expected behavior**  
Render the 404 page, with a 404 http status code.

## To reproduce
1. Take the Kirby Starter Kit
2. Add the following to the config:
```
    'routes' => [
        [
            'pattern' => 'test',
            'action'  => function () {
                throw new NotFoundException();
            }
        ]
    ]
````
3. Go to http://localhost/test

Or alternatively
1. Take the Kirby Starter Kit
2. Edit `site/controllers/notes.php` to include `throw new NotFoundException();` right at the beginning of the returned function.
3. Go to http://localhost/notes

Or even worse
1. Take the Kirby Starter Kit
2. Go for example to http://localhost/media/pages/photography/trees/1

In this last case, the NotFoundException is thrown from Kirby internally with the message "The thumbnail configuration could not be found".

I'm not sure if I should report a separate bug, because you can basically enter anything starting with `http://localhost/media/pages/photography/trees/...` and it will log a NotFoundException, even though it should just render a normal 404 page without logging anything.

## Your setup
**Kirby Version**  
Latest version 4.2.0

 - OS: Linux
 - Browser: Firefox
 - Version: 126

**Comment by afbora** [comment]
About media link issue is solved in https://github.com/getkirby/kirby/issues/6446 (will be available 4.3.0)

I was also realized that `NotFoundException` returns 500 http header instead 404 in that issue. I'm not sure this is bug or not.

**Comment by distantnative** [comment]
It's somehow expected as Kirby is only catching the `Kirby\Exception\ErrorPageException` once it enters page rendering (which involves the controller). However, I understand that `NotFoundException` might be expected to do the same. Let's see what the rest of the team says.
